packages/agreement-email-sender/src/index.ts

```diff
@@ -0,0 +1,62 @@
+import { runConsumer } from "kafka-iam-auth";
+import { EachMessagePayload } from "kafkajs";
+import {
+  ReadModelRepository,
+  agreementTopicConfig,
+  decodeKafkaMessage,
+  emailManagerConfig,
+  initEmailManager,
+  kafkaConsumerConfig,
+  logger,
+  readModelWriterConfig,
+} from "pagopa-interop-commons";
+import { AgreementEvent } from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import { selfcareV2Client } from "pagopa-interop-selfcare-v2-client";
+import { readModelServiceBuilder } from "./services/readModelService.js";
+import { sendAgreementEmail } from "./services/agreementEmailSenderService.js";
+
+const config = kafkaConsumerConfig();
+const readModelConfig = readModelWriterConfig();
+const topicsConfig = agreementTopicConfig();
+const emailConfig = emailManagerConfig();
+const emailManager = initEmailManager(emailConfig);
+
+const readModelService = readModelServiceBuilder(
+  ReadModelRepository.init(readModelConfig)
+);
+
+export async function processMessage({
+  message,
+}: EachMessagePayload): Promise<void> {
+  const decodedMessage = decodeKafkaMessage(message, AgreementEvent);
+  const loggerInstance = logger({
+    serviceName: "agreement-email-sender",
+    eventType: decodedMessage.type,
+    eventVersion: decodedMessage.event_version,
+    streamId: decodedMessage.stream_id,
+    correlationId: decodedMessage.correlation_id,
+  });
+  loggerInstance.debug(decodedMessage);
+
+  match(decodedMessage).with(
+    { event_version: 2, type: "AgreementActivated" },
+    { event_version: 2, type: "AgreementSubmitted" },
+    async ({ data: { agreement } }) => {
+      if (agreement) {
+        await sendAgreementEmail(
+          agreement,
+          readModelService,
+          selfcareV2Client,
+          emailManager
+        );
+      } else {
+        loggerInstance.error(
```

Comment:
No specific considerations, I think it is done like that because I did the same in `eservice-descriptors-archiver`. Fixing it by throwing an error in both